### PR TITLE
Fix masked quantities in RecurrentPPO 

### LINKED ISSUE #75: [question] Question regarding RecurrentPPO and the magnitude of approx_kl / clip_fraction / entropy_loss
While using `RecurrentPPO`, I noticed rather high values for `approx_kl` and `clip_fraction`. On further investigation it also seems that the `entropy_loss` indicates that the policy does not start from a (mostly) random policy. 

More concretely, I used `RecurrentPPO` and `PPO` (as a reference) to solve the LunarLander-v2 environment. Since this has an action space of four discrete actions we expect the starting (random) policy to have an entropy close to $-\log{1/4} \approx 1.39$. Which happens with `PPO`, but  the `RecurrentPPO` run it starts off lower, which seems odd to me. 

But more apparent is the difference in magnitude of the `approx_kl` for the two runs. 

**Question**
Is the magnitude difference of the `approx_kl` and starting point of the `entropy_loss` expected behaviour?

**Code example**
I used the following (pseudo) code to perform the test, while not one-to-one, I tried to get the model architectures to be roughly similar.

```python
vec_env = DummyVecEnv([lambda: Monitor(gym.make("LunarLander-v2"), filename = None) for _ in range(32)])
vec_env = VecNormalize(vec_env, norm_reward = True, gamma = .999)

policy_kwargs = dict( net_arch = [64, 64])
lstm_policy_kwargs = dict( net_arch = [64],
    lstm_hidden_size = 64, n_lstm_layers = 1, shared_lstm = True, enable_critic_lstm = False)

model = (Recurrent)PPO('Mlp(Lstm)Policy', vec_env, policy_kwargs = (lstm_)policy_kwargs,
    n_steps = 512,  batch_size = 128, n_epochs = 4,
    ent_coef = 0.01, vf_coef = .5, gamma = 0.999, gae_lambda = .98,
    normalize_advantage = True, 
    verbose = 1, tensorboard_log = PATH + '/tensorboard')

```

**TensorBoard results**
Here the orange run corresponds to `PPO`, while the blue run corresponds to `RecurrentPPO`

Screenshot 2022-06-06 at 15 24 50
Screenshot 2022-06-06 at 15 25 23

### PR DESCRIPTION
**Pull request by rnederstigt**
## Description

Changing the calculation of masked quantities from, 

```python
masked_mean = th.mean(values * mask)
```
to
```python
# Convert mask from float to boole
mask = mask > 1e-8
masked_mean = th.mean(values[mask])
```

## Context
closes #75 
- [x] I have raised an issue to propose this change ([required](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/master/CONTRIBUTING.md))

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] The functionality/performance matches that of the source (**required** for new training algorithms or training-related features).
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have included an example of using the feature (*required for new features*).
- [ ] I have included baseline results (**required** for new training algorithms or training-related features).
- [ ] I have updated the documentation accordingly.
- [x] I have updated the changelog accordingly (**required**).
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by Miffyli** [comment]
Hey,  thanks for the PR! I will let @araffin comment on the logic. Meanwhile, please complete the TODO list in the PR template (e.g., changelog and contributors) :)

File: sb3_contrib/ppo_recurrent/ppo_recurrent.py [hunk 1]
```diff
@@ -378,6 +378,9 @@ def train(self) -> None:
                     # Convert discrete action from float to long
                     actions = rollout_data.actions.long().flatten()
 
+                # Convert mask from float to boole
```
Comment (araffin) [inline sb3_contrib/ppo_recurrent/ppo_recurrent.py:381]:
```suggestion
                # Convert mask from float to bool
```

(use either Bool or Boolean)

File: sb3_contrib/ppo_recurrent/ppo_recurrent.py [hunk 2]
```diff
@@ -393,22 +396,19 @@ def train(self) -> None:
...
                 ratio = th.exp(log_prob - rollout_data.old_log_prob)
 
                 # clipped surrogate loss
                 policy_loss_1 = advantages * ratio
                 policy_loss_2 = advantages * th.clamp(ratio, 1 - clip_range, 1 + clip_range)
-                # Mask padded sequences
-                policy_loss_1 = policy_loss_1 * rollout_data.mask
-                policy_loss_2 = policy_loss_2 * rollout_data.mask
-                policy_loss = -th.min(policy_loss_1, policy_loss_2).mean()
+                policy_loss = -th.mean(th.min(policy_loss_1, policy_loss_2)[mask])
```
Comment (araffin) [inline sb3_contrib/ppo_recurrent/ppo_recurrent.py:407]: mask should be applied directly to `policy_loss_1` and `policy_loss_2`, no?

**Review by araffin — COMMENTED** [review]
Thanks for the PR =)

Only minor comments, please update the changelog too ;)

File: sb3_contrib/ppo_recurrent/ppo_recurrent.py (hunk 2 above)
Comment (rnederstigt) [reply sb3_contrib/ppo_recurrent/ppo_recurrent.py:407]: That's true, `policy_loss_1` and `policy_loss_2` are not strictly masked with zeros (due to the standardisation of the advantages and the ratios equating to ones for the masked indexes), but since we select only the valid indexes in `policy_loss` these are neglected anyway.

**Comment by araffin** [comment]
Please update the changelog and we are good to go =) (I did a quick check and it does solve all the mentioned issues without loss of performance).

PS: as you used master branch of your fork, I cannot push edits to it directly, please use another one next time ;)

**Comment by rnederstigt** [comment]
@araffin I updated the changelog. 

> PS: as you used master branch of your fork, I cannot push edits to it directly, please use another one next time ;)

I will keep that in mind for next time!
